rundeckapp/grails-app/services/rundeck/services/UserService.groovy

```diff
@@ -162,7 +123,7 @@ class UserService {
      * @deprecated
      */
     public storeFilterPref(String username, pref){
-        def User u = findOrCreateUser(username)
+        def RdUser u = findOrCreateUser(username)
```

Comment:
this method attempts to write to the `filterPref` field on the user object, so it will no longer work, it seems you will have to call `update` with the `userDataProvider` to store the data